rpc_component/schemata.py

```diff
@@ -134,20 +135,58 @@ def is_version_ids_unique(releases):
     }
 )
 
-component_metadata_schema = Schema(
+dependencies_schema = Schema(
     {
-        "dependencies": And(
+        Optional("dependencies"): And(
             [
                 {
                     "name": And(str, len),
                     "constraints": constraints_schema,
                 },
             ],
             is_value_unique("name"),
-        ),
+        )
+    }
+)
+
+artifacts_file_schema = Schema(
+    {
+        "type": "file",
+        "source": And(str, len),
+        Optional("dest"): And(str, len),
+        Optional("expire_after"): And(Use(int), lambda n: n > 0)
```

Comment:
I would rather it could only be one type, this potentially causes a pain point due to the way each is handled as well as potentially causing the input to differ from the output depending on internal representation.

Reply:
Fair enough, done.